code/modules/admin/verbs/freeze.dm

```diff
@@ -38,18 +38,18 @@ GLOBAL_LIST_EMPTY(frozen_atom_list) // A list of admin-frozen atoms.
 		GLOB.frozen_atom_list += src
 
 		var/obj/effect/overlay/adminoverlay/AO = new
-		if(skip_overlays)
+		if(!skip_overlays)
 			overlays += AO
 
 		anchored = TRUE
 		admin_prev_sleeping = AmountSleeping()
-		frozen = AO
+		frozen = TRUE
 		PermaSleeping()
 
 	else
 		GLOB.frozen_atom_list -= src
 
-		if(skip_overlays)
+		if(!skip_overlays)
 			overlays -= frozen
```

Comment:
It seems remove_overlay() only work with carbon/human? Next commit frozen will being swapped back to `AO`. And yeah, overlays are cut with all of those in their update_stats and stuff but I feel like it would still be a bad idea to remove removing the overlay here, even if its not needed, in case something weird happened.

Reply:
`remove_overlay` and likewise `add_overlay` are defined on atom so they should work.